src/pybind/mgr/cephadm/module.py

```diff
@@ -2503,6 +2565,135 @@ def upgrade_stop(self):
         self.event.set()
         return trivial_result('Stopped upgrade to %s' % target_name)
 
+    def save_spec(self, json_spec: dict) -> None:
+        """
+        Attempts to save a ServiceSpec. Raises `OrchestratorError` if
+        an entry with the identifier already exist.
+
+        There are not two ways to manipulate the service_specs stored in the mon_store
+        1) Using a global definition (i.e. `ceph orch apply -i <specs_file.yaml>`)
+            This will usually contain multiple definitions of services and daemons.
+
+        2) Using the CLI for specific service types (i.e. `ceph orch apply rgw 3 --realm foo --zone bar`)
+
+        If these entries interfere, one would overwrite the other and vice versa.
+        Having the mentioned safeguard in place will prevent overwrites from two different sources.
```

Comment:
```suggestion
         Raises `OrchestratorError` if an entry with the same named identifier already exist to prevent overwrites from two different sources.
```